src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -280,15 +287,103 @@ private boolean parseNodes(List<Node> nodes) {
           continue;
         }
 
-        if (cpeUriCheck(cpeMatch)) {
+        if (projectCheck(cpeMatch, references)) {
           return true;
         }
       }
 
-      if (parseNodes(node.getChildren())) {
+      if (parseNodes(node.getChildren(), references)) {
         return true;
       }
     }
     return false;
   }
+
+  /**
+   * Checks if the input {@link URI} is same as the project's URL.
+   * 
+   * @param referenceUrl of type {@link URI}.
+   * @return True if the input URL matches the project's URL. Otherwise false.
+   */
+  private boolean checkUrl(URI referenceUrl) {
+    if (referenceUrl.getHost().equals(project.url().getHost())
+        && referenceUrl.getScheme().equals(project.url().getProtocol())
+        && checkProjectUrl(referenceUrl.getPath())) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Extract {@link URI} from the input string URL.
+   * 
+   * @param url Input string URL.
+   * @return type {@link URI} or an {@link Optional#empty()}.
+   */
+  private Optional<URI> getUrl(String url) {
+    try {
+      return Optional.ofNullable(new URI(url));
+    } catch (URISyntaxException e) {
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Matches the reference URL with the project's URL.
+   * 
+   * @param path The path of the URL.
+   * @return True if the project URL matches with the reference URL. Otherwise False.
+   */
+  private boolean checkProjectUrl(String path) {
+    String[] splitPath = path.split("/");
+
+    if (splitPath.length >= 3 && splitPath[1].equals(project.organization().name())
```

Comment:
Would this work with a path without leading `/`? It may be better to trim leading `/`. Can this method be static?